modules/travel_pay/app/controllers/travel_pay/v1/claims_controller.rb

```diff
@@ -0,0 +1,190 @@
+      # @raise [Common::Exceptions::BadRequest] If the request is invalid
+      # @raise [Common::Exceptions::ResourceNotFound] If the claim is not found
+      def show
+        unless Flipper.enabled?(:travel_pay_view_claim_details, @current_user)
+          message = 'Travel Pay Claim Details unavailable per feature toggle'
+          raise Common::Exceptions::ServiceUnavailable, message:
+        end
+
+        begin
+          claim = claims_service.get_claim_by_id(params[:id])
+        rescue Faraday::Error => e
+          TravelPay::ServiceError.raise_mapped_error(e)
+        rescue ArgumentError => e
+          raise Common::Exceptions::BadRequest, message: e.message
+        end
+
+        if claim.nil?
+          raise Common::Exceptions::ResourceNotFound, message: "Claim not found. ID provided: #{params[:id]}"
+        end
+
+        render json: claim, status: :ok
+      end
+
+      # Creates a new claim with mileage expense
+      #
+      # @param appointment_datetime [String] The datetime of the appointment
+      # @return [JSON] Created claim with status 201
+      # @raise [Common::Exceptions::ServiceUnavailable] If feature toggle is disabled
+      # @raise [Common::Exceptions::BadRequest] If the request is invalid
+      # @raise [Common::Exceptions::ResourceNotFound] If the appointment is not found
+      # @raise [Common::Exceptions::InternalServerError] If there's a service error
+      def create
+        unless Flipper.enabled?(:travel_pay_submit_mileage_expense, @current_user)
+          message = 'Travel Pay mileage expense submission unavailable per feature toggle'
+          Rails.logger.error(message:)
+          raise Common::Exceptions::ServiceUnavailable, message:
+        end
+
+        begin
+          Rails.logger.info(message: 'SMOC transaction START')
+
+          appt_id = get_appt_or_raise(params['appointment_datetime'])
+          claim_id = get_claim_id(appt_id)
+
+          Rails.logger.info(message: "SMOC transaction: Add expense to claim #{claim_id.slice(0, 8)}")
+          expense_service.add_expense({ 'claim_id' => claim_id, 'appt_date' => params['appointment_datetime'] })
+
+          Rails.logger.info(message: "SMOC transaction: Submit claim #{claim_id.slice(0, 8)}")
+          submitted_claim = claims_service.submit_claim(claim_id)
+
+          Rails.logger.info(message: 'SMOC transaction END')
+        rescue ArgumentError => e
+          raise Common::Exceptions::BadRequest, detail: e.message
+        rescue Faraday::ClientError, Faraday::ServerError => e
+          raise Common::Exceptions::InternalServerError, exception: e
+        end
+
+        render json: submitted_claim, status: :created
+      end
+
+      private
+
+      # Validates the X-Client-GUID header and looks up the corresponding client number
+      #
+      # @raise [Common::Exceptions::BadRequest] If the header is missing or invalid
+      def validate_client_guid
```

Comment:
Note: The magic happens here.

Reply:
As a heads up, we're rapidly moving on to our "complex claims" and "mobile app SMOC" work, where we'll be adding more controllers, integrating with the mobile app, etc. It might be good to move this from the claims controller to maybe the application controller, or at least a more generic place.